content/articles/object-serialization-and-deserialization-in-java/index.md

```diff
@@ -0,0 +1,402 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /object-serialization-and-deserialization-in-java/
+title: Serialization and deserialization in Java
+description: This article explains the concept of serialization and deserialization in the context of Java.
+author: grace-wanjiru
+date: 2021-09-27T00:00:00-14:30
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+ - url: /engineering-education/object-serialization-and-deserialization-in-java/hero.png
+   alt: Serialization and deserialization diagram
+---
+In this tutorial, I'll explain how serialization works and how we can implement it. I will also include examples of how we can serialize a data structure using various Java concepts in doing so.
+<!--more-->
+
+You can read a deserialized object from a file and reconstruct it in memory. This utilizes the entity's metadata and bytes that constitute its contents.
+
+### Table of contents
+
+- [The concept of Serialization and Deserialization](#the-concept-of-serialization-and-deserialization)
+- [Characteristics of serialization](#characteristics-of-serialization)
+- [Benefits of serialization](#benefits-of-serialization)
+- [How to serialize and deserialize with the help of examples](#how-to-serialize-and-deserialize-with-the-help-of-examples)
+- [How to use inheritance to serialize an object in Java](#how-to-use-inheritance-to-serialize-an-object-in-java)
+- [Serialization with aggregation](#serialization-with-aggregation)
+- [Serialization with static data members](#serialization-with-static-data-members)
+- [Transient members](#transient-members)
+
+### The concept of serialization and deserialization
+
+The idea of **serialization in Java** is a byte-stream representation of an entity's state. All the entity's data is inside the stream of bits.
+
+**Deserialization** is the reversal of serialization, in which a byte data type stream is turned back to a memory object. The best thing about both techniques is that they're both JVM-agnostic, which means you can serialize on one JVM and de-serialize on another.
+
+### Characteristics of serialization
+
+- Machine Independent: On a separate device you can deserialize any encapsulated item.
+- Inheritance: When a parent class is serializable, all its subclasses become serializable as well.
+
+#### Why serialization?
+
+- To transfer objects through a network.
+- To keep Java objects in memory.
+- To save Java objects in files.
+
+### Benefits of serialization
+
+- It's used to organize troops (by traveling the status of a network attribute).

```

Comment:
I have no idea what you mean by *troops* here. Please revise this.

Reply:
Again, please revise this sentence: `It's used to organize troops (by traveling the status of a network attribute).` I have no idea what you mean by this. What is meant by *troops* and *traveling*?